lib/internal/http2/util.js

```diff
@@ -360,8 +360,7 @@ function isIllegalConnectionSpecificHeader(name, value) {
     case HTTP2_HEADER_TRANSFER_ENCODING:
       return true;
     case HTTP2_HEADER_TE:
-      const val = Array.isArray(value) ? value.join(', ') : value;
-      return val !== 'trailers';
+      return value.toString() !== 'trailers';
```

Comment:
`toString()` isn't really necessary here. If an array slips through then it won't match anyway. Should just be `value !== 'trailers'`.

Reply:
removed `toString()` in the updated commit